spec/components/shared/environment_banner_component_spec.rb

```diff
@@ -0,0 +1,33 @@
+require "rails_helper"
+
+RSpec.describe Shared::EnvironmentBannerComponent, type: :component do
+  describe "#render" do
+    before do
+      allow(Rails.configuration).to receive(:app_role).and_return(ActiveSupport::StringInquirer.new(app_role))
+    end
+
+    context "when app_role is production" do
+      let(:app_role) { "production" }
+
+      it "does not render" do
+        expect(subject.render?).to eq(false)
+      end
+    end
+
+    context "when app_role is unknown" do
```

Comment:
if we do a `render_inline` somewhere then we can do things like `expect(subject).to be_rendered` vice versa, but I'm happy enough if you just want to test the method itself instead